src/operator/src/req_convert/insert/row_to_region.rs

```diff
@@ -49,10 +50,36 @@ impl<'a> RowToRegion<'a> {
         let mut region_request = Vec::with_capacity(requests.inserts.len());
         let mut instant_request = Vec::with_capacity(requests.inserts.len());
         for request in requests.inserts {
+            let Some(rows) = request.rows else { continue };
+
             let table_id = self.get_table_id(&request.table_name)?;
-            let requests = Partitioner::new(self.partition_manager)
-                .partition_insert_requests(table_id, request.rows.unwrap_or_default())
-                .await?;
+            let table_route = self
+                .partition_manager
+                .find_physical_table_route(table_id)
+                .await
+                .context(FindTablePartitionRuleSnafu {
+                    table_name: &request.table_name,
+                })?;
+            let requests = if let Some(region_id) = match table_route.region_routes.split_first() {
+                Some((first, rest)) => {
+                    if rest.is_empty() {
+                        Some(first.region.id)
+                    } else {
+                        None
+                    }
+                }
+                None => Some(RegionId::new(table_id, 0)),
+            } {
+                vec![InsertRequest {
+                    region_id: region_id.as_u64(),
```

Comment:
Does it work well in the metric engine? It seems to be the physical region ID.

Reply:
Now the codes use the table info to find the singular region id, must work with logical table.